examples/tests/valid/resource.w

```diff
@@ -47,3 +47,5 @@ new cloud.Function(inflight () => {
   assert(s == "counter is: 1");
   assert(bucket.list().length == 1);
 }) as "test";
+
+resource Empty {}
```

Comment:
Could you add some test cases covering if this works with inheritance and classes?

```ts
class T1 {
 init(stuff: string) { }
}

class T2 extends T1 {
}

let x = new T2();
          //^ this should be a type error
let y = new T2("yay");
```

Reply:
Depending on how tricky this is, we could start by only allowing `init() {}` to be omitted when the class/resource doesn't `extends` another class/resource